notifier/notify_windows.go

```diff
@@ -2,351 +2,342 @@ package main
 
 import (
 	"fmt"
-	"io"
-	"net"
-	"os/exec"
-	"path/filepath"
-	"runtime"
-	"strings"
 	"sync"
+	"unsafe"
 
-	"github.com/Microsoft/go-winio"
+	"github.com/safing/portbase/info"
 	"github.com/safing/portbase/log"
-	"github.com/safing/portbase/utils/osdetail"
-	"golang.org/x/text/encoding/unicode"
-)
+	"github.com/safing/portmaster/updates/helper"
 
-var (
-	notifierPath                  string
-	notifierPathMutex             sync.Mutex
-	notificationsEnabledForThisOS bool // Are Notifications even enabled? (not on Windows Versions < 8); no locking done since only written in init()
+	"golang.org/x/sys/windows"
 )
 
+type NotificationID int64
+
 const (
-	pipeName          = "\\\\.\\pipe\\portmasterNotifierToast"
-	appID             = "io.safing.portmaster"
-	notificationTitle = "Portmaster"
+	appName     = "Portmaster"
+	company     = "Safing ICS Technologies GmbH"
+	productName = "Portmaster"
+	subProduct  = "notifier"
 )
 
-type actionCallback struct {
-	action string
-	button string // actionCallback.button is NOT garanteed to exist in the Callback. Therefore make sure to handle empty String
-	id     string
+// WinNotify holds the DLL handle.
+type NotifierLib struct {
+	sync.RWMutex
+
+	dll *windows.DLL
+
+	initialize           *windows.Proc
+	isInitialized        *windows.Proc
+	createNotification   *windows.Proc
+	deleteNotification   *windows.Proc
+	addButton            *windows.Proc
+	setImage             *windows.Proc
+	showNotification     *windows.Proc
+	hideNotification     *windows.Proc
+	setActivatedCallback *windows.Proc
+	setDismissedCallback *windows.Proc
+	setFailedCallback    *windows.Proc
 }
 
-type cmdArgs []string
+var (
+	lib            *NotifierLib
+	libInitialized bool
```

Comment:
If guard could be used concurrently, consider using https://github.com/tevino/abool instead.